caveclient/jsonservice.py

```diff
@@ -142,28 +144,36 @@ def get_neuroglancer_info(self, ngl_url=None):
         handle_response(response, as_json=False)
         return json.loads(response.content)
 
-    def get_state_json(self, state_id):
+    def get_state_json(
+        self,
+        state_id: Union[int, str],
+    ) -> dict:
         """Download a Neuroglancer JSON state
 
         Parameters
         ----------
-        state_id : int
-            ID of a JSON state uploaded to the state service.
+        state_id : int or string
+            ID of a JSON state uploaded to the state service. If a string, treats this as the url to the json state.
 
         Returns
         -------
         dict
             JSON specifying a Neuroglancer state.
         """
-        url_mapping = self.default_url_mapping
-        url_mapping["state_id"] = state_id
-        url = self._endpoints["get_state"].format_map(url_mapping)
+        url = None
+        try:
```

Comment:
what is the value error this is catching? just whether the state ID can become an int?

Reply:
Yeah, exactly. I wanted to be generous about allowing strings for the integers, since that can crop up in various pathways.